android/src/main/java/com/mattermost/networkclient/helpers/Helper.kt

```diff
@@ -212,6 +212,13 @@ fun formUrlString(baseUrl: String, endpoint: String): String {
     return baseUrl.toHttpUrlOrNull()!!.newBuilder().addPathSegments(endpoint.trim { c -> c == '/' }).build().toString()
 }
 
+/**
+ *
+ */
+fun String.trimSlashes(): String {
+    return this.trim { c -> c == '/' }
```

Comment:
I've been wondering about this - I think we might have to revisit the comment from before about using specific parts of the URL strictly; something like `scheme://host:port` - I think a `baseUrl` is too open to interpretation; so when a the `baseUrl` is passed in, we parse it into it's URL fragments and use them; this way there's no ambiguity around ports/trailing slashes etc.

The `trimSlashes()` was because of bad input; we really should be strict with input rather than try to capture and handle edge cases...

Reply:
How about something like https://github.com/mattermost/react-native-network-client/pull/53/commits/06404391a7d03e7f76ad2d9d89631fcf21f48a78 - it seems to work pretty well and sets a good default.

Reply:
The problem about being strict with scheme://host:port is that you are leaving out the possibility of using a virtual folder or subpath as part of your baseUrl (scheme://host:port/path) for example https://shaz.com/mattermost